gammapy/estimators/metadata.py

```diff
@@ -0,0 +1,256 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import logging
+from typing import Optional
+import numpy as np
+from scipy import stats
+from astropy.coordinates import SkyCoord
+from pydantic import ValidationError, validator
+from gammapy.utils.metadata import CreatorMetaData, MetaData
+
+__all__ = ["FluxMetaData"]
+
+SEDTYPE = ["dnde", "flux", "eflux", "e2dnde", "likelihood"]
+FPFORMAT = ["gadf-sed", "lightcurve"]
+STANDARD_KEYS = [
+    "SED_TYPE",
+    "SEDTYPEI",
+    "UL_CONF",
+    "N_SIGMA",
+    "NSIGMAUL",
+    "STSTHUL",
+    "NSIGMSEN",
+    "TARGETNA",
+    "TARGETPO",
+    "OBS_IDS",
+    "DATASETS",
+    "INSTRU",
+]
+
+log = logging.getLogger(__name__)
+
+
+class FluxMetaData(MetaData):
+    """Metadata containing information about the FluxPoints and FluxMaps.
+
+    Parameters
+    ----------
+    sed_type : {"dnde", "flux", "eflux", "e2dnde", "likelihood"}, optional
+        SED type.
+    sed_type_init : {"dnde", "flux", "eflux", "e2dnde", "likelihood"}, optional
+        SED type of the initial data.
+    ul_conf : float, optional
+        Confidence level used for the upper limit computation.
+    n_sigma : float, optional
+        Significance threshold above which upper limits should be used.
+    n_sigma_ul : float, optional
+        Sigma number used to compute the upper limits.
+    sqrt_ts_threshold_ul : float, optional
+        Threshold on the square root of the likelihood value above which upper limits should be used.
+    n_sigma_sensitivity : float, optional
+        Sigma number for which the flux sensitivity is computed
+    # gti : `~gammapy.data.gti`, optional
+    #     used Good Time Intervals.
+    target_name : str, optional
+        Name of the target.
+    target_position : `~astropy.coordinates.SkyCoord`, optional
+        Coordinates of the target.
+    obs_ids : list of int, optional
+        ID list of the used observations.
+    dataset_names : list of str, optional
+        Name list of the used datasets.
+    instrument : str, optional
+        Name of the instrument.
+    creation : `~gammapy.utils.CreatorMetaData`, optional
+        The creation metadata.
+    optional : dict, optional
+        additional optional metadata.
+
+    Note: these quantities are serialized in FITS header with the keywords stored in STANDARD_KEYS
+    """
+
+    sed_type: Optional[str]  # Are these 6 fields really optional?
+    sed_type_init: Optional[str]
+    ul_conf: Optional[float]
+    n_sigma: Optional[float]
+    n_sigma_ul: Optional[float]
+    sqrt_ts_threshold_ul: Optional[float]
+    n_sigma_sensitivity: Optional[float]
+    # gti: Optional[GTI]
+    target_name: Optional[str]  # Are these 2 fields really optional?
+    target_position: Optional[SkyCoord]
+    obs_ids: Optional[list[int]]
+    dataset_names: Optional[list[str]]  # Are these 2 fields really optional?
```

Comment:
```suggestion
    dataset_names: Optional[List[str]]  # Are these 2 fields really optional?
```

Reply:
If they are not optional you won't be able to read external or previously generated flux points.